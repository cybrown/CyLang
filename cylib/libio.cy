void extern putchar(int c);
void extern putbyte(byte c);
void extern putshort(short c);
void extern putint(int c);
void extern putlong(long c);
void extern putfloat(float f);
void extern putdouble(double d);
void extern puts(byte[] str);
void extern printf(byte[] str);

void print(byte c) {
	putbyte(c);
}

void print(short c) {
	putshort(c);
}

void print(int c) {
	putint(c);
}

void print(long c) {
	putlong(c);
}

void print(float f) {
	putfloat(f);
}

void print(double d) {
	putdouble(d);
}

void print(byte[] str) {
	printf(str);
}

void print(bool c) {
	print(c ? "true" : "false");
}
