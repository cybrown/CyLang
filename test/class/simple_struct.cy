import libio;

class Foo {
	int a;
	int b;
	int c;
	byte[] str;
}

class Bar {
	Foo my_foo2;
	Foo my_foo;
}

void Foo_setStr(Foo this, byte[] s) {
	this.str = s;
}

void Bar_setFooStr(Bar this, byte[] s) {
	this.my_foo.str = s;
}

Bar my_bar = Bar();
my_bar.my_foo = Foo();

Foo_setStr(my_bar.my_foo, "Hello from an Object !\n");

my_bar.my_foo.a = 13;
my_bar.my_foo.a = 40;
my_bar.my_foo.b = 2;
my_bar.my_foo.c = my_bar.my_foo.a + my_bar.my_foo.b;

print(my_bar.my_foo.str);
print(my_bar.my_foo.c + 52);
putchar(10);
