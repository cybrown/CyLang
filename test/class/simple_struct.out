Hello from an Object !
94
